### PR TITLE
Feat/monitoring documentation

### PR DESCRIPTION
**Pull request by oxedions**
Monitoring documentation.

From @sobrase

### CODE REVIEW

**Comment by oxedions** [comment]
Contains monitoring documentation.
Still perfectible, but nice start.

Also took the opportunity to solve 2 minors issues on the dhcp_server and pxe_stack readmes found during html static build.

File: resources/documentation/monitoring/grafana.rst [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+as you can see, in one case, you get the results in different panels, and in the
+other case you get the results in the same panel.
+
+Field
+"""""
+
+Unit
+****
+
+Self explanatory, choose the unit, min, max and the display name for the values.
```
Comment (osmocl) [inline resources/documentation/monitoring/grafana.rst:287]: Self-explanatory

File: resources/documentation/monitoring/ipmi_exporter.rst [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+management nodes) under */etc/ipmi_exporter/ipmi_config.yml*
+
+By default, it should look like this:
+
+.. code-block:: yaml
+
+  modules:
+
+  equipment_compute_SEQ2_CER:
+    user: super
```
Comment (osmocl) [inline resources/documentation/monitoring/ipmi_exporter.rst:101]: Please change this user/password.

**Review by osmocl — CHANGES_REQUESTED** [review]
Some minors advices.

File: resources/documentation/monitoring/grafana.rst [hunk 3]
```diff
@@ -119,20 +119,20 @@ Transform
...
 For example :
 
-.. image:: monitoring/capture/grafana/transformExmemple.PNG
+.. image:: /monitoring/capture/grafana/transformExmemple.PNG
    :width: 50 %
 
 Here, we have 3 queries, but if you make no transform, it will look like this:
 
-.. image:: monitoring/capture/grafana/transformExmemple3.PNG
+.. image:: /monitoring/capture/grafana/transformExmemple3.PNG
```
Comment (btravouillon) [inline resources/documentation/monitoring/grafana.rst:135]: Should the file be named transformExample3.png?

File: resources/documentation/monitoring/grafana.rst [hunk 4]
```diff
@@ -119,20 +119,20 @@ Transform
...
+.. image:: /monitoring/capture/grafana/transform1.PNG
    :width: 30 %
 
 Mainly useful when using graphs. Allows you to show the things you want in the
 table, by reducing, filtering, joining metrics, and organizing fields.
 
 For example :
 
-.. image:: monitoring/capture/grafana/transformExmemple.PNG
+.. image:: /monitoring/capture/grafana/transformExmemple.PNG
```
Comment (btravouillon) [inline resources/documentation/monitoring/grafana.rst:130]: Should the file be named transformExample.png?

File: resources/documentation/monitoring/grafana.rst [hunk 5]
```diff
@@ -212,7 +212,7 @@ Legend
 Legend related options, you can show the legend as a table, add min, max, avg,
 current values.
 
-.. image:: monitoring/capture/grafana/LegendExemple.PNG
+.. image:: /monitoring/capture/grafana/LegendExemple.PNG
```
Comment (btravouillon) [inline resources/documentation/monitoring/grafana.rst:215]: Should the file be named LegendExample.png?

File: resources/documentation/monitoring/grafana.rst [hunk 6]
```diff
@@ -0,0 +1,391 @@
...
+Manual installation
+^^^^^^^^^^^^^^^^^^^
+
+There should be a package for grafana under the bluebanquise repository.
+
+By running the following command:
+
+.. code-block:: text
+
+  yum install grafana
```
[the remark this answers is not in the document]
Comment (btravouillon) [reply resources/documentation/monitoring/grafana.rst:30]: The stack supports both EL 7 and 8. There is a symlink `/usr/bin/yum -> dnf-3`  in EL 8. AFAIK, there is no `dnf` in EL 7, so I would keep `yum` in the documentation.

File: resources/documentation/monitoring/ipmi_exporter.rst (hunk 2 above)
Comment (btravouillon) [reply resources/documentation/monitoring/ipmi_exporter.rst:101]: Why?

Comment (osmocl) [reply resources/documentation/monitoring/ipmi_exporter.rst:101]: sensitive information

File: resources/documentation/monitoring/grafana.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (osmocl) [reply resources/documentation/monitoring/grafana.rst:30]: Yes, I know the symlink, however sometimes dnf command does not work porperly in RHEL7 boxes.
I propose:
````
dnf install grafana # (RHEL8)
yum install grafana # (RHEL7)
````
@oxedions could you confirm it please ?

[the remark this answers is not in the document]
Comment (btravouillon) [reply resources/documentation/monitoring/grafana.rst:30]: Any sysadmin can translate yum to dnf.

[the remark this answers is not in the document]
Comment (btravouillon) [reply resources/documentation/monitoring/grafana.rst:30]: > [...] however sometimes dnf command does not work porperly in RHEL7 boxes.

So use `yum` command :smiley:

File: resources/documentation/monitoring/ipmi_exporter.rst (hunk 2 above)
Comment (btravouillon) [reply resources/documentation/monitoring/ipmi_exporter.rst:101]: These are default credentials available in the documentation of the hardware platform. If this is considered sensitive, we might consider using other kind of equipments for the documentation.

File: resources/documentation/monitoring/grafana.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (osmocl) [reply resources/documentation/monitoring/grafana.rst:30]: I updated my comment 😃

File: resources/documentation/monitoring/ipmi_exporter.rst (hunk 2 above)
Comment (oxedions) [reply resources/documentation/monitoring/ipmi_exporter.rst]: Solved.

**Comment by hmeAtos** [comment]
Beware with the resources/documentation/monitoring/capture/grafana/Capture.PNG screenshot
You can take this picture [](url)
Capture

regards

**Comment by sobrase** [comment]
Fixed all issues in feat/monitoring_documentation

File: resources/documentation/monitoring/grafana.rst [hunk 7]
```diff
@@ -0,0 +1,391 @@
...
+A dashboard is made of Query and Alerts.
+
+Query
+^^^^^
+
+metrics field
+"""""""""""""
+
+This section is where you put Prometheus metric queries.
+See the Prometheus section of the doc for more info about the metrics
```
[the remark this answers is not in the document]
Comment (btravouillon) [reply resources/documentation/monitoring/grafana.rst]: @sobrase you forgot to replace **info** with **information** in your last commits.

File: resources/documentation/monitoring/grafana.rst (hunk 1 above)
Comment (btravouillon) [reply resources/documentation/monitoring/grafana.rst:276]: still true

File: resources/documentation/roles.rst [hunk 8]
```diff
@@ -7,11 +7,14 @@ This part is an agregations of all the **BlueBanquise** roles readme.
 CORE
 ====
 
+.. include:: roles/core/access_control/readme.rst
```
Comment (btravouillon) [inline resources/documentation/roles.rst:10]: while relevant, these changes are not related to monitoring. A separate PR would have been appropriate.

Comment (oxedions) [reply resources/documentation/roles.rst:10]: My bad, I did it on the wrong branch. 🙁 

Lets keep it this way this time.

File: roles/addons/prometheus_client/readme.rst [hunk 9]
```diff
@@ -21,4 +72,3 @@ Changelog
 
 * 1.0.0: Documentation. johnnykeats <johnny.keats@outlook.com>
 * 1.0.0: Role creation. johnnykeats <johnny.keats@outlook.com>
```
Comment (btravouillon) [inline roles/addons/prometheus_client/readme.rst:74]: typo?

File: roles/addons/prometheus_server/readme.rst [hunk 10]
```diff
@@ -15,6 +15,34 @@ Alertmanager is available at http://localhost:9093
 
 Node_exporter is available at http://localhost:9100
 
+By default, role will inherit from values set in its defaults folder.
+You may wish to update these values to your needs, as these values set the
+different timings usd by Prometheus.
```
Comment (btravouillon) [inline roles/addons/prometheus_server/readme.rst:20]: **used**

File: roles/addons/prometheus_server/readme.rst [hunk 11]
```diff
@@ -15,6 +15,34 @@ Alertmanager is available at http://localhost:9093
...
+    scrape_interval: 1m
+    evaluation_interval: 2m
+    alertmanager:
+      group_wait: 1m
+      group_interval: 10m
+      repeat_interval: 3h
+
+And tune according to your needs.
+
+Note also that the prometheus_server role will try to read the prometheus_client
```
Comment (btravouillon) [inline roles/addons/prometheus_server/readme.rst:37]: .. note:: ?

File: resources/documentation/monitoring/introduction.rst [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+============
+
+This monitoring documentation has multiple aims:
+
+* Provide an overview of Prometheus and the different exporters.
+* Provide instructions on how to add exporters to nodes, and how you could add your own exporters.
+* Provide instructions on alerting configuration.
+* Provide an overview of Grafana and the different dashboard.
+
+That is why this documentation will be built around those section:
```
Comment (btravouillon) [inline resources/documentation/monitoring/introduction.rst:11]:
```suggestion
This documentation is built around those section:
```

File: resources/documentation/monitoring/node_exporter.rst [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+  node_vmstat_pswpout 532036
+  # HELP process_cpu_seconds_total Total user and system CPU time spent in seconds.
+  # TYPE process_cpu_seconds_total counter
+  process_cpu_seconds_total 20272.88
+  # HELP process_max_fds Maximum number of open file descriptors.
+  # TYPE process_max_fds gauge
+  process_max_fds 1024
+
+There are a lot of information given by node_exporter. It is a very heavy
+exporter, so that is why it is not running on the compute nodes.
```
Comment (btravouillon) [inline resources/documentation/monitoring/node_exporter.rst:42]: Is it a recommendation? (Anyone can run it on the compute nodes with the appropriate playbook).

File: resources/documentation/monitoring/prometheus.rst [hunk 14]
```diff
@@ -0,0 +1,594 @@
...
+service name.
+
+.. note::
+  As you can see, you can also add the scrape_interval (which is how
+  often the metrics get scraped), and the scrape_timeout (which represents how
+  long until a scrape request times out).
+
+.. note::
+  If you want to add exporters, make sure your package contains the
+  binary and the .service file, put preferably under /usr/local/bin and
```
Comment (btravouillon) [inline resources/documentation/monitoring/prometheus.rst:118]: why preferably in /usr/local/bin?

Comment (sobrase) [reply resources/documentation/monitoring/prometheus.rst:118]: all packaged exporters provided by BB have their binaries located under /usr/local/bin, so we know where they all are

File: resources/documentation/monitoring/node_exporter.rst (hunk 13 above)
Comment (sobrase) [reply resources/documentation/monitoring/node_exporter.rst:42]: yes it is a recommendation, I will rephrase that

File: resources/documentation/monitoring/prometheus.rst (hunk 14 above)
Comment (btravouillon) [reply resources/documentation/monitoring/prometheus.rst:118]: rpm -ql <package> can provide the location of the files.
I understand however.

File: roles/addons/prometheus_client/readme.rst (hunk 9 above)
Comment (oxedions) [reply roles/addons/prometheus_client/readme.rst:74]: yes. solved.

File: roles/addons/prometheus_server/readme.rst (hunk 10 above)
Comment (oxedions) [reply roles/addons/prometheus_server/readme.rst:20]: good catch !

File: roles/addons/prometheus_server/readme.rst (hunk 11 above)
Comment (oxedions) [reply roles/addons/prometheus_server/readme.rst:37]: yes, done.

File: resources/documentation/monitoring/introduction.rst (hunk 12 above)
Comment (oxedions) [reply resources/documentation/monitoring/introduction.rst:11]: done

**Comment by oxedions** [comment]
Ready !
